crates/interledger-http/src/lib.rs

```diff
@@ -30,6 +31,99 @@ pub trait HttpStore: Clone + Send + Sync + 'static {
     /// received on the incoming HTTP request.
     fn get_account_from_http_token(
         &self,
+        username: &str,
         token: &str,
     ) -> Box<dyn Future<Item = Self::Account, Error = ()> + Send>;
 }
+
+// Helpers for parsing authorization methods
+pub const BEARER_TOKEN_START: usize = 7;
+pub const BASIC_TOKEN_START: usize = 6;
+
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct Auth {
+    username: String,
+    password: String,
+}
+
+impl Auth {
+    pub fn new(username: &str, password: &str) -> Self {
+        Auth {
+            username: username.to_owned(),
+            password: password.to_owned(),
+        }
+    }
+
+    pub fn username(&self) -> &str {
+        &self.username
+    }
+
+    pub fn password(&self) -> &str {
+        &self.password
+    }
+
+    pub fn parse(s: &str) -> Result<Self, ()> {
+        let s = s.replace("%3A", ":"); // When the token is received over the wire, it is URL encoded. We must decode the semi-colons.
+        if s.starts_with("Basic") {
+            Auth::parse_basic(&s[BASIC_TOKEN_START..])
+        } else if s.starts_with("Bearer") {
+            Auth::parse_bearer(&s[BEARER_TOKEN_START..])
+        } else {
+            Auth::parse_text(&s)
+        }
+    }
+
+    fn parse_basic(s: &str) -> Result<Self, ()> {
+        let decoded = base64::decode(&s).map_err(|_| ())?;
+        let text = std::str::from_utf8(&decoded).map_err(|_| ())?;
+        Auth::parse_text(text)
+    }
+
+    // Currently, we use bearer tokens in a non-standard way, where they each have a
+    // username and a password in them. In the future, we will deprecate Bearer auth
+    // for accounts.
+    fn parse_bearer(s: &str) -> Result<Auth, ()> {
+        Auth::parse_text(s)
+    }
+
+    fn parse_text(text: &str) -> Result<Self, ()> {
+        let parts = &mut text.split(':');
+        let username = match parts.next() {
+            Some(part) => part.to_owned(),
+            None => return Err(()),
+        };
+        let password = match parts.next() {
+            Some(part) => part.to_owned(),
+            None => return Err(()),
+        };
+        Ok(Auth { username, password })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parses_correctly() {
+        assert_eq!(
+            Auth::parse("Basic ZXZhbjpzY2h3YXJ6").unwrap(),
+            Auth::new("evan", "schwarz")
```

Comment:
```suggestion
            Auth::new("evan", "schwartz")
```
Not a fan of using my name in the tests cause it makes it look like I'm super conceited unless you look at `git blame` to see who included it